File: src/types/export.ts
export type ExportMode = 'copy' | 'move';
export type ExportCategory = 'selected' | 'rejected' | 'favorite' | 'skipped';

export interface ExportRequest {
  sessionId: string;
  categories: ExportCategory[];
  destinationRoot: string;
  mode: ExportMode;
  organizeByCategory: boolean;
  verifyHashes: boolean;
}

export interface ExportFailure {
  path: string;
  reason: string;
}

export interface ExportReport {
  totalRequested: number;
  succeeded: number;
  failed: ExportFailure[];
  missingSources: string[];
  skippedExisting: string[];
  verifiedHashes: number;
  durationMs: number;
}

File: src-tauri/src/models/export.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportRequest {
    pub session_id: String,
    pub categories: Vec<String>, // "selected"|"rejected"|"favorite"|"skipped"
    pub destination_root: String,
    pub mode: String,              // "copy"|"move"
    pub organize_by_category: bool,
    pub verify_hashes: bool,
}

#[derive(Debug, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ExportFailure {
    pub path: String,
    pub reason: String,
}

#[derive(Debug, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ExportReport {
    pub total_requested: usize,
    pub succeeded: usize,
    pub failed: Vec<ExportFailure>,
    pub missing_sources: Vec<String>,
    pub skipped_existing: Vec<String>,
    pub verified_hashes: usize,
    pub duration_ms: u64,
}
